### PR TITLE
Added 'effective' user field.

### PR DESCRIPTION
**Pull request by matthewscherer**
## Change Summary
This adds the "effective" field to the user field of security events. 

### Sample values

Sample document:
```
{
    "@timestamp": "2024-03-28T13:57:06.9120679Z",
    "agent": {
        "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        "type": "endpoint",
        "version": "8.14.0-SNAPSHOT"
    },
    "data_stream": {
        "dataset": "endpoint.events.security",
        "namespace": "default",
        "type": "logs"
    },
    "ecs": {
        "version": "8.10.0"
    },
    "elastic": {
        "agent": {
            "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
        }
    },
    "event": {
        "action": "log_on",
        "category": [
            "authentication",
            "session"
        ],
        "created": "2024-03-28T13:57:06.9120679Z",
        "dataset": "endpoint.events.security",
        "id": "NUJr8jg5OdZe9mSa++++++EF",
        "kind": "event",
        "module": "endpoint",
        "outcome": "success",
        "sequence": 1234,
        "type": [
            "start"
        ]
    },
    "host": {
        "architecture": "x86_64",
        "hostname": "desktop-1tk590j",
        "id": "dabadaba-0000-0000-0000-000000000000",
        "ip": [
            "192.168.157.189",
            "fe80::4687:22c0:187:a16e",
            "169.254.58.69",
            "fe80::8d58:8a17:2e45:3163",
            "127.0.0.1",
            "::1"
        ],
        "mac": [
            "00-0c-29-1c-6d-2b",
            "7c-b5-66-a8-17-35"
        ],
        "name": "desktop-1tk590j",
        "os": {
            "Ext": {
                "variant": "Windows 11 Pro"
            },
            "family": "windows",
            "full": "Windows 11 Pro 21H2 (10.0.22000.2538)",
            "kernel": "21H2 (10.0.22000.2538)",
            "name": "Windows",
            "platform": "windows",
            "type": "windows",
            "version": "21H2 (10.0.22000.2538)"
        }
    },
    "message": "Endpoint security event",
    "process": {
        "Ext": {
            "ancestry": [
                "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTE3NDkyLTE3MTE2MzQxMzIuMzUwMTI1ODAw",
                "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTkyMjAtMTcxMTEyNDcyOS4zNjIzNDgwMA=="
            ],
            "code_signature": [
                {
                    "exists": true,
                    "status": "trusted",
                    "subject_name": "Python Software Foundation",
                    "trusted": true
                }
            ]
        },
        "code_signature": {
            "exists": true,
            "status": "trusted",
            "subject_name": "Python Software Foundation",
            "trusted": true
        },
        "entity_id": "YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTE1MjA0LTE3MTE2MzQxMzIuMzY4ODU1NTAw",
        "executable": "C:\\Users\\Default.DESKTOP-1TK590J\\AppData\\Local\\Programs\\Python\\Python38\\python.exe",
        "name": "C:\\Users\\Default.DESKTOP-1TK590J\\AppData\\Local\\Programs\\Python\\Python38\\python.exe"
    },
    "user": {
        "domain": "DESKTOP-1TK590J",
        "effective": {
            "domain": "DESKTOP-1TK590J",
            "name": "EafTestLogon80589"
        },
        "id": "S-1-5-21-1749029863-1064264096-968553628-1001",
        "name": "Default"
    }
}
```

### For mapping changes:

- [ ] I ran `make` after making the schema changes, and committed all changes
- [ ] If these field(s) are "exception"-able, I made a companion PR to Kibana adding it (see [Readme](https://github.com/elastic/endpoint-package#exceptionable))
- [ ] If this is a `metadata` change, I also updated both transform destination schemas to match

### For Transform changes:

- [ ] The new transform successfully starts in Kibana
- [ ] The corresponding transform destination schema was updated if necessary

### CODE REVIEW

**Review by gabriellandau — COMMENTED** [review]
Changes LGTM.  Could you add the new fields to the [sample security event](https://github.com/elastic/endpoint-package/blob/main/package/endpoint/data_stream/security/sample_event.json) used in unit tests?

**Review by gabriellandau — APPROVED** [review]
LGTM, thanks

File: package/endpoint/data_stream/security/fields/fields.yml [hunk 1]
```diff
@@ -886,3 +886,26 @@
           default_field: false
       description: Short name or login of the user.
       example: a.einstein
+    - name: user.effective.id
```
Comment (pzl) [inline package/endpoint/data_stream/security/fields/fields.yml:889]: This field is already nested under the `user` property, so including `user` again in the keyname gives you `user.user.effective.id` as the result

File: package/endpoint/data_stream/security/fields/fields.yml [hunk 2]
```diff
@@ -886,3 +886,26 @@
...
+          default_field: false
+      description: Short name or login of the effective user.
+      example: a.einstein
+    - name: user.effective.domain
+      level: extended
+      type: keyword
+      ignore_above: 1024
+      description: 'Name of the directory the effective user is a member of.
+
+        For example, an LDAP or Active Directory domain name.'
```
Comment (pzl) [inline package/endpoint/data_stream/security/fields/fields.yml:911]: email, hash, and full_name are missing here

File: package/endpoint/data_stream/security/sample_event.json [hunk 3]
```diff
@@ -85,8 +85,13 @@
...
     "user": {
-        "domain": "NT AUTHORITY",
-        "name": "SYSTEM",
-        "id": "S-1-5-18"
+        "domain": "DESKTOP-1TK590J",
+        "effective": {
+            "domain": "DESKTOP-1TK590J",
+            "name": "EafTestLogon80589",
+            "id": "S-1-5-18"
+        },
```
Comment (pzl) [inline package/endpoint/data_stream/security/sample_event.json:93]: can we have sample values for `full_name`, `email`, and `hash` too?

File: package/endpoint/docs/README.md [hunk 4]
```diff
@@ -2720,6 +2720,9 @@ sent by the endpoint.
 | user.hash | Unique user hash to correlate information for a user in anonymized form. Useful if `user.id` or `user.name` contain confidential information and cannot be used. | keyword |
 | user.id | Unique identifier of the user. | keyword |
 | user.name | Short name or login of the user. | keyword |
+| user.user.effective.domain | Name of the directory the effective user is a member of. For example, an LDAP or Active Directory domain name. | keyword |
```
Comment (pzl) [inline package/endpoint/docs/README.md:2723]: This picked up on the duplicate `user` in the path

File: package/endpoint/data_stream/security/fields/fields.yml (hunk 1 above)
Comment (pzl) [reply package/endpoint/data_stream/security/fields/fields.yml:889]: was this hand-edited?

you can `rm` this file if you want `make` to remake it

**Comment by matthewscherer** [comment]
@pzl Thanks for the help! The tests are passing and I added some of those fields to the test json.

**Review by pzl — APPROVED** [review]

**Pull request merged**